homeassistant/components/mqtt/client.py

```diff
@@ -209,12 +221,15 @@ async def async_subscribe(
             translation_domain=DOMAIN,
             translation_placeholders={"topic": topic},
         ) from exc
-    return await mqtt_data.client.async_subscribe(
-        topic,
-        msg_callback,
-        qos,
-        encoding,
-    )
+    client = mqtt_data.client
+    if not client.connected and not mqtt_config_entry_enabled(hass):
+        raise HomeAssistantError(
```

Comment:
Seems we need to adjust a test here.

Reply:
Looks like we have no test for this and it was accidentally covered

Reply:
added coverage for this